src/js/actions/export.js

```diff
@@ -467,38 +498,93 @@ define(function (require, exports) {
+            return Promise.resolve();
         }
 
-        return _exportService.promptForFolder(_lastFolderPath || "~")
-            .bind(this)
-            .then(
-                function (baseDir) {
-                    _lastFolderPath = baseDir;
-
-                    // Iterate over the exports map, find the associated layer, test of "exportEnabled"
-                    var exportArray = documentExports.getLayersWithExports(document, undefined, true)
-                        .flatMap(function (layer) {
-                            return documentExports.getLayerExports(layer.id)
-                                .map(function (asset, index) {
-                                    return _exportAsset.call(this, document, layer, index, asset, baseDir);
-                                }, this);
-                        }, this);
+        // helper to perform the export after a directory has been determined
+        var exportToDir = function (baseDir) {
+            _lastFolderPath = baseDir;
 
-                    return Promise.all(exportArray.toArray());
-                },
-                function (err) {
-                    // promptForFolder rejected, probably just user-canceled
-                    if (err.message && err.message.startsWith("cancelError: cancel")) {
-                        log.warn("Prompt for folder failed, and it wasn't a simple 'cancel'");
-                        throw new Error("Failed to open an OS folder chooser dialog: " + err.message);
-                    }
-                    return Promise.resolve();
+            var layersList;
+            if (layerIDs) {
+                // TODO push this logic into DocumentExports model, and filter first on layerIDs
+                layersList = documentExports.getLayersWithExports(document).filter(function (layer) {
+                    return layerIDs.includes(layer.id);
                 });
+            } else {
+                layersList = documentExports.getLayersWithExports(document, undefined, true);
+            }
+
+            var layerIdList = collection.pluck(layersList, "id");
+
+            return _setAssetsRequested.call(this, document.id, layerIdList).then(function () {
+                // Iterate over the layers, find the associated export assets, and export them
+                var exportArray = layersList.flatMap(function (layer) {
+                    return documentExports.getLayerExports(layer.id)
+                        .map(function (asset, index) {
+                            return _exportAsset.call(this, document, layer, index, asset, baseDir);
+                        }, this);
+                }, this);
+
+                return Promise.all(exportArray.toArray());
+            });
+        }.bind(this);
+
+        // prompt for folder and then export to the result.
+        // resolve immediately if no folder is returned
+        return _exportService.promptForFolder(_lastFolderPath || "~")
```

Comment:
@ktaki When you test this, can you verify that with clean preferences the default folder-chooser location is the user's home directory on Windows?